navigation-compose/src/main/kotlin/com/freeletics/khonshu/navigation/compose/NavHost.kt

```diff
@@ -53,13 +57,17 @@ public fun NavHost(
     destinations: Set<NavDestination>,
     deepLinkHandlers: Set<DeepLinkHandler> = emptySet(),
     deepLinkPrefixes: Set<DeepLinkHandler.Prefix> = emptySet(),
+    navController: NavHostController = rememberNavController(),
     destinationChangedCallback: ((BaseRoute) -> Unit)? = null,
 ) {
     val context = LocalContext.current
 
     val overlayNavigator = remember { OverlayNavigator() }
     val customActivityNavigator = remember(context) { CustomActivityNavigator(context) }
-    val navController = rememberNavController(overlayNavigator, customActivityNavigator)
+
+    navController.navigatorProvider.addNavigator(overlayNavigator)
+    navController.navigatorProvider.addNavigator(customActivityNavigator)
+
```

Comment:
Two statements can throw `"Navigator ... is replacing an already attached ..."`. https://github.com/freeletics/khonshu/pull/580 has fixed the similar problem.

Reply:
I also notice they are `side-effect`s, and should be in Compose Effect(s)

Reply:
@hoc081098 I don't think the exception can happen here anymore. `addNavigator` has an early return if the given navigator is the same as the `previousNavigator`. Since both navigators are being remembered they will be the same even after a re-composition. Before your PR the `CustomActivityNavigator` would always be newly instantiated

Reply:
@gabrielittner I got it, thank you for clarifying 🙏